app/models/concerns/digital_object/assets/featured_region.rb

```diff
@@ -0,0 +1,77 @@
+module DigitalObject::Assets::FeaturedRegion
+  extend ActiveSupport::Concern
+
+  def featured_region=(region)
+    region = region.first if region.is_a?(Array)
+    @fedora_object.clear_relationship(:region_featured)
+    @fedora_object.add_relationship(:region_featured, region, true) if region.present?
+  end
+
+  def featured_region
+    rels = @fedora_object&.relationships(:region_featured)
+    current_value = rels.first.object if rels.present?
+    current_value
+  end
+
+  def region_selection_event=(event_data)
+    event_data = event_data.first if event_data.is_a?(Array)
+    @fedora_object.clear_relationship(:region_selection_event)
+    if event_data.present?
+      event_data['updatedAt'] ||= Time.now.utc.iso8601
+      @fedora_object.add_relationship(:region_selection_event, JSON.generate(event_data), true)
+    end
+  end
+
+  def region_selection_event
+    rels = @fedora_object&.relationships(:region_selection_event)
+    current_value = rels.first.value if rels.present?
+    JSON.load(current_value) || { 'updatedBy' => I18n.t('email.automatic_process'), 'updatedAt' => updated_at.iso8601 }
+  end
+
+  def asset_image_width
+    return nil unless @fedora_object
+    width_val = @fedora_object.rels_int.relationships(@fedora_object.datastreams['content'], :image_width).first&.object
+    width_val ||= @fedora_object.relationships(:image_width).first
+    width_val.to_s.to_i
+  end
+
+  def asset_image_length
+    return nil unless @fedora_object
+    width_val = @fedora_object.rels_int.relationships(@fedora_object.datastreams['content'], :image_length).first&.object
+    width_val ||= @fedora_object.relationships(:image_length).first
```

Comment:
```suggestion
    width_val ||= @fedora_object.relationships(:image_length).first
    width_val = @fedora_object.rels_int.relationships(@fedora_object.datastreams['content'], :image_length).first&.object
```

Proposing check of rels-ext first because going forward, Derivativo will set the rels-ext value, and may do this on resources that previously only had a rels-int value.

Reply:
Done and rebased!